blog/2023-07-31-quarterly-earnings-chatbot/index.md

```diff
@@ -0,0 +1,227 @@
+---
+slug: quarterly-earnings-chatbot
+title: Build A Q&A Chatbot For Quarterly Earnings Calls
+authors: [het]
+tags: [llm, self-hosted, prem, open-source, fintech, langchain, vicuna-7b, weaviate, vector-store, streamlit]
+---
+
+![Prem Banner](./banner.png)
+
+<head>
+  <meta name="twitter:image" content="./banner.jpg"/>
+</head>
+
+## Financial background information
+
+Publicly tradable companies are required to show investors how they are doing from a business and financial standpoint. Most of the information that investors are looking for can be found in a document called the 10-Q (Quarterly earnings report) and the 10-K (Annual earnings report ).
+
+Usually, companies have an earnings call live with their investors and then publish their earnings reports afterward. So the people that get on the earnings call have a head start and get to know the information in real-time.
+
+> What if I told you that there was a way to capture all of the data from a live earnings call without having to sit through the entire thing?
+
+## Using AI to process financial data
+
+There is an open-source AI model called Whisper Tiny that can take audio and convert it into text in a matter of seconds. This model is small enough to run on your CPU, yet it has the speed and accuracy to process even the largest of audio files.
+
+Going back to the earnings call, we can use Whisper Tiny to convert the audio into text which we can process later on. We can then build a chatbot to "chat" with the text related to a company's finances. 
+
+Using this technology, investors would be able to ask the chatbot questions like Where has the revenue grown most this year? or What was the net revenue for the quarter? This allows people to analyze the performance of a publicly traded company in near real-time without having to wait for the earnings report to get published.
+
+## The application
+
+Here are the steps required to build this app:
+1. Run the Whisper Tiny model locally
+2. Send the audio file to the model and get the converted text
+3. Use Langchain to break the text into chunks
+4. Store the chunks inside a vector database
+5. Query the data stored in the vector database
+6. Chat with the data using an open-source LLM
+
+There are quite a few technologies here and it can be difficult to run them manually. This is where Prem AI comes in. Prem AI allows people to easily deploy open-source tools and models without having to deal with any infrastructure.
+
+
+We can run the Whisper Tiny model, Weaviate vector database, open-source embeddings tool, and the open-source Vicuna LLM all through Prem AI. 
+
+### Let's get started!
+
+<hr />
+
+## Step 1: Install python dependencies
```

Comment:
these steps should map 1:1 with the numbered list above.

We're completely missing (1) "how to run a model (Whisper Tiny) locally with Prem AI".